code/modules/atmospherics/machinery/components/unary_devices/passive_vent.dm

```diff
@@ -20,36 +20,19 @@
 /obj/machinery/atmospherics/components/unary/passive_vent/process_atmos()
 	..()
 
-	var/active = FALSE
-
 	var/datum/gas_mixture/external = loc.return_air()
 	var/datum/gas_mixture/internal = airs[1]
-	var/external_pressure = external.return_pressure()
-	var/internal_pressure = internal.return_pressure()
-	var/pressure_delta = abs(external_pressure - internal_pressure)
-
-	if(pressure_delta > 0.5)
-		if(external_pressure < internal_pressure)
-			var/air_temperature = (external.temperature > 0) ? external.temperature : internal.temperature
-			var/transfer_moles = (pressure_delta * external.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			var/datum/gas_mixture/removed = internal.remove(transfer_moles)
-			external.merge(removed)
-		else
-			var/air_temperature = (internal.temperature > 0) ? internal.temperature : external.temperature
-			var/transfer_moles = (pressure_delta * internal.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			transfer_moles = min(transfer_moles, external.total_moles() * internal.volume / external.volume)
-			var/datum/gas_mixture/removed = external.remove(transfer_moles)
-			if(isnull(removed))
-				return
-			internal.merge(removed)
-
-		active = TRUE
-
-	active = internal.temperature_share(external, OPEN_HEAT_TRANSFER_COEFFICIENT) ? TRUE : active
-
-	if(active)
-		air_update_turf()
-		update_parents()
+
+	var/internal_volume_fraction = internal.volume/(internal.volume + external.volume)
+
+	var/datum/gas_mixture/holding = new
+	holding.merge(internal.remove_ratio(1))
+	holding.merge(external.remove_ratio(1))
+	internal.merge(holding.remove_ratio(internal_volume_fraction))
+	external.merge(holding)
+
+	air_update_turf()
+	update_parents()
```

Comment:
If see only on code compactness, it's nice, but can be compacted more.
you can use one of `gas_mixture` internal or external, instead of creating holding.
```suggestion
	external.merge(internal.remove_ratio(1))
	internal.merge(external.remove_ratio(internal.volume/external.volume))

	air_update_turf()
	update_parents()
```

If think about load cost, it can be increased, i'm not sure.

Reply:
Only
```internal.volume/(internal.volume+external.volume)``` 
jumps out at me.